### ISSUE TITLE
`deno task`: cannot run tsx

### ISSUE THREAD

**Issue by lowlighter** [denoland/deno#27166]
Version: Deno 2.1.2

This is the same error as #20625, but not in the same context.

Trying to create an alias for `npx tsx --test` (to test codebase on node js) but for some reason when executed through the task runner there a weird error (see stacktrace below)
```json
{
  "tasks": {
    "test:node": "npx tsx --help"
  }
}
```
```
Task test:node npx tsx --help
tsx

Node.js runtime enhanced with esbuild for loading TypeScript & ESM

Usage:
  tsx [flags...] [script path]
  tsx <command>

Commands:
  watch        Run the script and watch for changes                                                                                                                      

Flags:
  -h, --help                     Show help                                                                                                                               
      --no-cache                 Disable caching                                                                                                                         
      --tsconfig <string>        Custom tsconfig.json path                                                                                                               
  -v, --version                  Show version                                                                                                                            

---------------------------------------------

error: unexpected argument '--require' found

  tip: to pass '--require' as a value, use '-- --require'

Usage: deno run [OPTIONS] [SCRIPT_ARG]...
```

There's a "deno run" mention so I assume there's some kind of shenanigans that alias/replace tsx.

If executed directly in bash the command works as expected

**Comment by marvinhagemeister** [comment]
The `--require` argument in node is used for loaders. The `tsx` package uses that to shim support for reading ts/tsx files directly in Node. Loaders are not supported in Deno, see https://github.com/denoland/deno/issues/8327 . It's been a long discussion if those should be added in Deno or not.

Meanwhile, the `tsx` command isn't necessary in Deno since Deno can run TS files natively. Maybe a workaround for your use case is to skip the `tsx` command. Not sure if that works in your particular project.

**Comment by lowlighter** [comment]
@marvinhagemeister 
The thing is that in my case, deno compatibility with tsx should be irrevelant, as tsx is *not* supposed to be actually be run by deno but by node (I just use `deno task` as a shell wrapper).

This is why I was wondering whether `npx`/`tsx` are treated differently by the deno task runner or not, because the same command in shell works (and also I don't understand why it prints a message about `deno run` when I did not ask to run deno in the command)

Here's a more complete extract of my `deno.jsonc` tasks so it may be easier to understand what I'm trying to achieve:
```json
"test": {
  "description": "Test code and print detailed coverage",
  "command": "cd $INIT_CWD && deno coverage --detailed --include=$INIT_CWD",
  "dependencies": ["test:deno", "test:node", "test:bun"]
},
"test:deno": {
  "description": "Test code (deno)",
  "command": "cd $INIT_CWD && deno test --allow-read --allow-env --allow-net --parallel --doc --clean --coverage"
},
"test:node": {
  "description": "Test code (node)",
  "command": "cd $INIT_CWD && npx tsx --test" // <-- this throw the error from the issue
},
"test:bun": {
  "description": "Test code (bun)",
  "command": "cd $INIT_CWD && bun test"
},
```

I have no issue to run my test for deno/bun (as you said, they both support ts natively so it's fine), but for node I need the `tsx` part to make it compatible

**Comment by marvinhagemeister** [comment]
Right, when running `deno task`, Deno will alias the `node` and `npx` binary to Deno equivalent commands. I don't think we have a flag to not do that, so the workaround for now is to keep invoking those commands with `npm run` or other npm package managers.

**Comment by lowlighter** [comment]
Ok thanks to your suggestion I was able to make it work by using `scripts: npx tsx --test` in the `package.json` and replace `test:node` by `npm test` in `deno.jsonc`, thanks a lot

Btw is the behavior you mentioned documented somewhere ? 
Because I didn't find anything in https://docs.deno.com/runtime/reference/cli/task/ hence my initial confusion, and it's really error prone if you don't know this "magic" aliasing

**Comment by bartlomieju** [comment]
@lowlighter I don't think it's mentioned in the docs, but probably should. Any chance you could open a PR for that in https://github.com/denoland/docs?

As for the issue itself - I guess the crux here is lack of "module loaders" support, that we are eye-balling for Q1 '25.
